### PR TITLE
Added IProcessor Interface and renamed ItemTracker

### PR DESCRIPTION
**Pull request by sdilts**
#### Changes
+ ItemTracker now uses the IProcessor interface to track items
+ This change was made so that ItemTracker could be tested
+ Can now interact with the processing code outside of the itemTracker

Please Review: @sagesmith-wf

### CODE REVIEW

File: src/main/java/csci432/ProcessController.java [hunk 1]
```diff
@@ -43,6 +47,18 @@ private void run() {
...
     /**
+     * Processes all items left in the queue
+     **/
+    public void flushQueue() {
+	while(!queue.isEmpty()) {
+	    //take is blocking equivalent
+	    BufferedImage item = queue.poll();
+	    processItem(item);
+	}
+
```
Comment (sagersmith8) [inline src/main/java/csci432/ProcessController.java]: remove newline

File: src/main/java/csci432/ProcessController.java [hunk 2]
```diff
@@ -10,12 +10,16 @@
  * of images that have been passed through Sigma-Delta and edge
  * detection algorithms.
  *
+ * @param T the type of the object that will process each item
```
Comment (sagersmith8) [inline src/main/java/csci432/ProcessController.java:13]: Probably not necessary anymore

File: src/test/java/csci432/ProcessControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import org.junit.Test;
+import java.awt.image.BufferedImage;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+public class ProcessControllerTest {
+
```
Comment (sagersmith8) [inline src/test/java/csci432/ProcessControllerTest.java]: remove newline

File: src/test/java/csci432/ProcessControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    public void enterItems(ProcessController proc, int numItems) {
+	for(int i = 0; i < numItems; i++) {
+	    BufferedImage b = new BufferedImage(500,500,
+						BufferedImage.TYPE_INT_ARGB);
+	    b.setRGB(i,0,checkColor);
+	    proc.insertItem(b);
+	}
+    }
+
+
```
Comment (sagersmith8) [inline src/test/java/csci432/ProcessControllerTest.java:25]: remove newline

File: src/test/java/csci432/ProcessControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+	}
+
+	public int getItemsGiven() {
+	    return itemsGiven;
+	}
+
+	public boolean getOrderCorrect() {
+	    return orderCorrect;
+	}
+
```
Comment (sagersmith8) [inline src/test/java/csci432/ProcessControllerTest.java]: Remove newline

File: src/test/java/csci432/ProcessControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+	public int getItemsGiven() {
+	    return itemsGiven;
+	}
+
+	public boolean getOrderCorrect() {
+	    return orderCorrect;
+	}
+	
+
```
Comment (sagersmith8) [inline src/test/java/csci432/ProcessControllerTest.java]: Remove newline

**Comment by sagersmith8** [comment]
Looks good other than style. Looks like you abstracted to an interface to make it easier to test

**Comment by sdilts** [comment]
Anything else? Is using an interface like this a good idea?

**Comment by sdilts** [comment]
I'm going to make it extend `Runnable` next. For the use case I have in mind, you need to communicate with the thread in some manner, and that will help

**Comment by sagersmith8** [comment]
Just don't worry about testing it too indepth yet, we haven't really solidified our design yet because we don't know if the libraries are going to work yet

**Pull request merged**